### PR TITLE
bump serde-json-core

### PR DESCRIPTION
**Pull request by jordens**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ categories = ["no-std", "config", "embedded", "parsing"]
 
 [dependencies]
 derive_miniconf = { path = "derive_miniconf" , version = "0.5" }
-serde-json-core = "0.4.0"
+serde-json-core = { git = "https://github.com/quartiq/serde-json-core", branch = "rj/maybeuninit" }
```
[the remark this answers is not in the document]
Comment (jordens) [reply Cargo.toml:17]: On our `Error`? What do we need that for?

[the remark this answers is not in the document]
Comment (ryan-summers) [reply Cargo.toml:17]: There's an explicit note about the derives for the bumped serde-json-core version

[the remark this answers is not in the document]
Comment (ryan-summers) [reply Cargo.toml:17]: https://github.com/quartiq/miniconf/blob/dc984a2a35f9287959f40ea5495dce32142e7d4a/src/lib.rs#L205

**Comment by jordens** [comment]
GH web UI seems lacking.
image

**Review by ryan-summers — APPROVED** [review]

**Pull request merged**